### PR TITLE
Adding support for AlternateKeyODataUriResolver

### PR DESCRIPTION
**Pull request by abkmr**
Adding support for AlternateKeyODataUriResolver
Corresponding updates in ODataUriResolverSetttings.

### CODE REVIEW

**Comment by xuzhg** [comment]
@abkmr The product codes look good to me. However, would you please add UT, FT, and E2E? Thanks

**Comment by xuzhg** [comment]
@abkmr How to set the alternate key by annotation on CLR models?

File: OData/test/UnitTest/System.Web.OData.Test/HttpConfigurationExtensionsTest.cs [hunk 1]
```diff
@@ -228,6 +228,20 @@ public void EnableEnumPrefixFree_Sets_EnumPrefixFreeFlag()
             Assert.True(resolverSetttings.EnumPrefixFree);
         }
 
+        [Fact]
+        public void EnableAlternateKeys_Sets_EnumPrefixFreeFlag()
```
Comment (xuzhg) [inline OData/test/UnitTest/System.Web.OData.Test/HttpConfigurationExtensionsTest.cs]: change the test case name

**Comment by xuzhg** [comment]
@abkmr Need E2E test cases and one simple page to introduce the alternate key usage. 

see: http://odata.github.io/WebApi/#04-14-capabilities-vocabulary-support

File: OData/test/UnitTest/System.Web.OData.Test/HttpConfigurationExtensionsTest.cs (hunk 1 above)
Comment (abkmr) [inline OData/test/UnitTest/System.Web.OData.Test/HttpConfigurationExtensionsTest.cs]: changed and updated PR.

**Comment by abkmr** [comment]
Added E2E test case. Will add the document as mentioned.

Thanks
Abhishek



On Thu, Jul 23, 2015 at 10:21 PM -0700, "Sam Xu" <notifications@github.com<mailto:notifications@github.com>> wrote:


@abkmr<https://github.com/abkmr> Need E2E test cases and one simple page to introduce the alternate key usage.

see: http://odata.github.io/WebApi/#04-14-capabilities-vocabulary-support

-

File: OData/test/E2ETest/WebStack.QA.Test.OData/UriParserExtension/AlternateKeysTest.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            configuration.EnsureInitialized();
+        }
+
+        [NuwaWebConfig]
+        public static void UpdateWebConfig(WebConfigHelper webConfig)
+        {
+            webConfig.AddRAMFAR(true);
+        }
+
+        public static TheoryDataSet<string, string, string> CaseInsensitiveCases
```
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/UriParserExtension/AlternateKeysTest.cs]: why also named as "CaseInsensitiveCases"?

File: OData/test/E2ETest/WebStack.QA.Test.OData/UriParserExtension/AlternateKeysTest.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
...ion>\r\n      </EntityType>\r\n      <EntityType Name=\"Order\">\r\n        <Key>\r\n          <PropertyRef Name=\"Id\" />\r\n        </Key>\r\n        <Property Name=\"Id\" Type=\"Edm.Int32\" Nullable=\"false\" />\r\n        <Property Name=\"Title\" Type=\"Edm.String\" />\r\n      </EntityType>\r\n      <EntityType Name=\"VipCustomer\" BaseType=\"WebStack.QA.Test.OData.UriParserExtension.Customer\">\r\n        <Property Name=\"VipProperty\" Type=\"Edm.String\" />\r\n      </EntityType>\r\n      <EnumType Name=\"Gender\">\r\n        <Member Name=\"Male\" Value=\"1\" />\r\n        <Member Name=\"Female\" Value=\"2\" />\r\n      </EnumType>\r\n    </Schema>\r\n    <Schema Namespace=\"Default\" xmlns=\"http://docs.oasis-open.org/odata/ns/edm\">\r\n      <EntityContainer Name=\"Container\">\r\n        <EntitySet Name=\"Customers\" EntityType=\"WebStack.QA.Test.OData.UriParserExtension.Customer\">\r\n          <NavigationPropertyBinding Path=\"Orders\" Target=\"Orders\" />\r\n        </EntitySet>\r\n        <EntitySet Name=\"Orders\" EntityType=\"WebStack.QA.Test.OData.UriParserExtension.Order\" />\r\n      </EntityContainer>\r\n    </Schema>\r\n  </edmx:DataServices>\r\n</edmx:Edmx>"},
```
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/UriParserExtension/AlternateKeysTest.cs]: You don't add cases to illustrate how to use alternate key?

File: OData/test/E2ETest/WebStack.QA.Test.OData/UriParserExtension/AlternateKeysTest.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        public async Task EnableAlternateKeysTest(string method, string requestPath, string comparisonValue)
+        {
+            var requestUri = string.Format("{0}/odata/{1}", this.BaseAddress, requestPath);
+            HttpRequestMessage request = new HttpRequestMessage(new HttpMethod(method), requestUri);
+            HttpResponseMessage response = await Client.SendAsync(request);
+
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            string responseContent = await response.Content.ReadAsStringAsync();
+
+            Assert.Equal(responseContent, comparisonValue);
```
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/UriParserExtension/AlternateKeysTest.cs]: (Expect, Actual)

File: OData/test/E2ETest/WebStack.QA.Test.OData/UriParserExtension/AlternateKeysTest.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                    { "Get", "Customers(1)/Name/$value", "Customer #1"}
```
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/UriParserExtension/AlternateKeysTest.cs]: Why test this?

**Comment by xuzhg** [comment]
Merged the related E2E test cases. #451. Thanks.

**Pull request closed without merging**